### PR TITLE
[OT171-22] Sign up validation

### PR DESCRIPTION
**Pull request by marianobattaglia**
Validation of email and password fields with regular expressions, and sign up button disabled when the fields dont respect the requirements.

### CODE REVIEW

**Comment by gapoj** [comment]
There are to many files in this PR that should not be here

File: SomosMasApp/SomosMasApp/App/Extensions/Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    // Regular expressions to validate fields
+    var isValidEmail: Bool {
+        NSPredicate(format: "SELF MATCHES %@", "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}").evaluate(with: self)
+    }
+    
+    var isValidPassword: Bool {
+        NSPredicate(format: "SELF MATCHES %@", "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$").evaluate(with: self)
+    }
+    
+    var isValidConfirmPassword: Bool {
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Extensions/Extensions.swift:21]: You don't need this and is against SOLID principles..... remember DRY(don't repeat yourself)

**Comment by marianobattaglia** [comment]
Resolved repeated validation fields and files uploaded by mistake were deleted

**Comment by marianobattaglia** [comment]
Two conflicts pushing files .gitignore and SceneDelegate corrected.

**Review by gapoj — APPROVED** [review]

**Pull request merged**